lib/files/node.ts

```diff
@@ -167,10 +171,23 @@ export abstract class Node {
 
 	/**
 	 * Get the file mime
-	 * There is no setter as the mime is not meant to be changed
 	 */
-	get mime(): string|undefined {
-		return this._data.mime
+	get mime(): string {
+		return this._data.mime || 'application/octet-stream'
+	}
+
+	/**
+	 * Set the file mime
+	 * Removing the mime type will set it to `application/octet-stream`
+	 */
+	set mime(mime: string|undefined) {
+		if (mime === undefined) {
+			mime = 'application/octet-stream'
+		}
+
+		validateData({ ...this._data, mime }, this._knownDavService)
+		this.updateMtime()
```

Comment:
changing the MIME is not changing the data within it, so I would not think the mtime is changed.

Reply:
Fair point!